packages/gatsby-source-filesystem/src/create-remote-file-node.js

```diff
@@ -3,112 +3,283 @@ const got = require(`got`)
 const crypto = require(`crypto`)
 const path = require(`path`)
 const { isWebUri } = require(`valid-url`)
+const Queue = require(`better-queue`)
 
 const { createFileNode } = require(`./create-file-node`)
 const cacheId = url => `create-remote-file-node-${url}`
 
+/********************
+ * Type Definitions *
+ ********************/
+
 /**
- * Index of promises resolving to File node from remote url
+ * @typedef {Redux}
+ * @see [Redux Docs]{@link https://redux.js.org/api-reference}
  */
-const processingCache = {}
 
-module.exports = ({ url, store, cache, createNode, auth = {} }) => {
-  // Check if we already requested node for this remote file
-  // and return stored promise if we did.
-  if (processingCache[url]) {
-    return processingCache[url]
+/**
+ * @typedef {GatsbyCache}
+ * @see gatsby/packages/gatsby/utils/cache.js
+ */
+
+/**
+ * @typedef {Auth}
+ * @type {Object}
+ * @property {String} htaccess_pass
+ * @property {String} htaccess_user
+ */
+
+/**
+ * @typedef {CRFNPayload}
```

Comment:
Any reason you shorten type prefix to CRFN? Not a blocker for me, just curious if we couldn't use full name here so anyone using it later don't have to decipher abbreviation.

Reply:
It just felt super long, particularly when using it in a param tag. I'll update it